### PR TITLE
feat(resolver): stream content

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: iroh-gateway/src/client.rs [hunk 1]
```diff
@@ -40,21 +48,23 @@ impl Client {
...
             .tts_file
             .set(start_time.elapsed().as_millis() as u64);
-        let n = res.len() as u64;
-        state
-            .metrics
-            .bytes_per_sec_out
-            .set(n / start_time.elapsed().as_secs().max(1));
-        state.metrics.bytes_streamed.inc_by(n);
-        Ok(res.into())
+        // let n = res.len() as u64;
```
Comment (Arqu) [inline iroh-gateway/src/client.rs:61]: you can drop this completely, we don't need this counter. Alternatively I'll do it on the next metrics cleanup

File: iroh-gateway/src/client.rs [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
 use crate::response::ResponseFormat;
 
-#[derive(Debug, Clone, Copy)]
-pub struct Client {}
+#[derive(Debug)]
+pub struct Client {
+    resolver: Resolver<iroh_rpc_client::Client>,
+}
+
+pub type PrettyStreamBody = StreamBody<ReaderStream<OutPrettyReader<iroh_rpc_client::Client>>>;
```
Comment (Arqu) [inline iroh-gateway/src/client.rs:18]: We might want to wrap this and hook in some metrics.

**Review by Arqu — APPROVED** [review]
Dope!

File: iroh-gateway/src/client.rs (hunk 2 above)
Comment (dignifiedquire) [reply iroh-gateway/src/client.rs:18]: as far as I understand axum, we probably want to add a generic middleware that does that for us

File: iroh-gateway/src/client.rs (hunk 1 above)
Comment (dignifiedquire) [reply iroh-gateway/src/client.rs]: removed

**Review by Arqu — APPROVED** [review]
🐰

**Pull request merged**